apps/notifications/src/doc/note-rendering.md

```diff
@@ -0,0 +1,105 @@
+				url={ this.props.note.url }
+			/>
+		);
+		break;
+	case 'comment':
+		body.push( <Comment key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'post':
+		body.push( <Post key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'reply':
+		replyBlock = <ReplyBlock key={ blockKey } block={ block.block } />;
+		break;
+	default:
+		body.push( p( html( block.block ) ) );
+		break;
+}
+```
+
+Depending on the item's type, a different block component is rendered - `User` for a representation of a user (for instance, in a notification of a user liking a particular post), `Comment` for a comment made on a blog post, `Post` for a blog post (i.e. on a blog the user is following), `ReplyBlock` for a reply to a comment(?).
+
+_n.b. - these 'blocks' are separate from the concept of 'blocks' in the Gutenberg block editor; a notification might have multiple blocks (for instance, multiple `User` blocks for multiple likes on a post), or it might have a single `Post` block, which then further renders Gutenberg blocks as part of the post content._
+
+These notification block components will be your entry point for any modifications you need to do to the way a particular notification is rendered to the user.
+
+For instance, if the user has received a notification about a post on a blog they are following, the component in question will be `src/panel/notifications/block-post.jsx`:
+
+```js
+import React from 'react';
+
+import { html } from '../indices-to-html';
+import { p } from './functions';
+
+const PostBlock = ( { block } ) => <div className="wpnc__post">{ p( html( block ) ) }</div>;
+
+export default PostBlock;
+```
+
+First, the block is passed to the `html()` function from `/src/panel/indices-to-html/index.js`, which processes the block's content information (making use of the `ranges` mentioned earlier) to split that information into chunks based on how they should be rendered. These chunks are then processed to reintroduce HTML tags back into the plain text content at the points indicated by the values in the `ranges` property.
+
+The result of this function (a string enhanced with HTML tag information) is then passed to `p()` in `src/panel/templates/functions.jsx`, which adds further HTML tag formatting based on line breaks and other special characters (such as backticks for code blocks).
+
+## Troubleshooting and Updating Renderings
+
+Local development of notifications can take place making use of the Calypso dev server, as any changes made to `src/apps/notifications` will be reflected in the Calypso local development environment automatically.
+
+However, as notifications can also be viewed in an iframe on non-Calypso sites using the masterbar (see https://github.com/Automattic/wp-calypso/edit/master/apps/notifications/README.md), any changes to the notifications sub-application should also be tested on a sandboxed WP.com site. To do this, run the following commands:
```

Comment:
What's the `masterbar`? Also I wonder if it's appropriate to add screenshots to READMEs. That's what I was planning to do.

Reply:
Our Jetpack/WP.com version of the logged-in admin bar:

<img width="956" alt="Screen Shot 2020-09-10 at 2 58 28 PM" src="https://user-images.githubusercontent.com/349751/92813499-1d362680-f376-11ea-8066-576ba5983fcf.png">

https://github.com/Automattic/jetpack/blob/865ea01b299b3926387f778de089cc4787602b73/modules/masterbar.php

Reply:
> Also I wonder if it's appropriate to add screenshots to READMEs

Sure, go for it – you could place them `/static/images`. We do avoid using huge amounts, really high resolution, or videos, to avoid bloating the repo's download size – in that case you could hang the image on a public service (like Cloudup), or commit it to WP.com.